src/framework/components/esmscript/component.js

```diff
@@ -461,6 +461,9 @@ class EsmScriptComponent extends Component {
         if (!ModuleClass.name || ModuleClass.name === '')
             throw new Error('Anonymous classes are not supported. Please use `class MyClass{}` as opposed to `const MyClass = class{}`');
 
+        if (this.moduleNameInstanceMap.has(ModuleClass.name))
+            throw new Error(`A esm script class called '${ModuleClass.name}' has already been added to this component.`);
```

Comment:
"An esm" :)

Also, it is probably a good idea to keep the same name, e.g. if we call it "ESM module script", then we should use the same one in all comments/docs.